src/systemctl/systemctl.c

```diff
@@ -2433,14 +2433,23 @@ static int unit_file_find_path(LookupPaths *lp, const char *unit_name, char **un
 
         STRV_FOREACH(p, lp->search_path) {
                 _cleanup_free_ char *path;
+                _cleanup_free_ char *lpath = NULL;
+                int r;
 
                 path = path_join(arg_root, *p, unit_name);
                 if (!path)
                         return log_oom();
 
-                if (access(path, F_OK) == 0) {
-                        *unit_path = path;
-                        path = NULL;
+                r = chase_symlinks(path, arg_root, &lpath);
+                if (r == -ENOENT)
+                        continue;
+                if (r == -ENOMEM)
+                        return log_oom();
+                if (r < 0)
+                        return log_error_errno(r, "Failed to access path '%s': %m", path);
+                if (r == 0) {
```

Comment:
I don't like this. It suggests to the reader that `r > 0` is a different case. But it's not. So this last `if` should go away.